server/pulp/plugins/rsync/publish.py

```diff
@@ -0,0 +1,522 @@
+            skip_fast_forward |= misc.strbool2bool(str_skip_ff)
+            if entry.get("result", "error") == "error":
+                skip_fast_forward = True
+
+        if self.last_published and isinstance(self.last_published, basestring):
+            last_published = dateutils.parse_iso8601_datetime(self.last_published)
+        else:
+            last_published = self.last_published
+        if last_published:
+            last_published = last_published.replace(tzinfo=None)
+
+        if self.last_deleted and isinstance(self.last_deleted, basestring):
+            last_deleted = dateutils.parse_iso8601_datetime(self.last_deleted)
+        else:
+            last_deleted = self.last_deleted
+        if last_deleted:
+            last_deleted = last_deleted.replace(tzinfo=None)
+
+        config_skip_ff = self.get_config().get("skip_fast_forward", False)
+        skip_fast_forward = skip_fast_forward | misc.strbool2bool(config_skip_ff)
+        force_refresh = self.get_config().get("force_refresh", False)
+
+        return last_published and ((last_deleted and last_published > last_deleted) or
+                                   not last_deleted) and not skip_fast_forward and\
+            not force_refresh
+
+    def create_date_range_filter(self, config):
+        """
+        Create a date filter based on start and end issue dates specified in the repo config. The
+        returned filter is a dictionary which can be used directly in a mongo query.
+
+        :param config: plugin configuration instance
+        :type  config: pulp.plugins.config.PluginCallConfiguration
+
+        :return: date filter dict with issued date ranges in mongo query format
+        :rtype:  dict
+        """
+        start_date = config.get(START_DATE_KEYWORD)
+        end_date = config.get(END_DATE_KEYWORD)
+        date_filter = None
+        if start_date and not isinstance(start_date, basestring):
+            start_date_str = dateutils.format_iso8601_datetime(start_date)
+        else:
+            start_date_str = start_date
+        if end_date and not isinstance(end_date, basestring):
+            end_date_str = dateutils.format_iso8601_datetime(end_date)
+        else:
+            end_date_str = end_date
+
+        if start_date and end_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$gte": start_date_str,
+                                                          "$lte": end_date_str}}
+        elif start_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$gte": start_date_str}}
+        elif end_date:
+            date_filter = {ASSOCIATED_UNIT_DATE_KEYWORD: {"$lte": end_date_str}}
+        return date_filter
+
+    def get_remote_repo_path(self):
+        """
+        Returns the full path to the published repository on remote server.
+        """
+        if "relative_url" in self.repo.notes and self.repo.notes["relative_url"]:
```

Comment:
Ideally this should come from the distributor config. The yum distributor does have the ability to look in the notes field to see if the importer left it a note about the feed's relative path, but even in that case, I think it saves the result in the distributor config and reads it from there. Reading the value from the notes field does not give the user a chance to change the value in the config.